samples/hello/src/com/microsoft/aad/adal/hello/MainActivity.java

```diff
@@ -281,12 +282,17 @@ public void onClickUseToken(View view) {
     }
 
     public void onClickClearTokens(View view) {
-        if (mAuthContext != null && mAuthContext.getCache() != null) {
-            displayMessage("Clearing tokens");
-            mAuthContext.getCache().removeAll();
-        } else {
-            textView1.setText("Cache is null");
-        }
+        try {
+			if (mAuthContext != null && mAuthContext.getCache() != null) {
+			    displayMessage("Clearing tokens");
+			    mAuthContext.getCache().removeAll();
+			} else {
+			    textView1.setText("Cache is null");
+			}
+		} catch (final UsageAuthenticationException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
The UsageAuthenticationException is thrown by getCache() which calls canSwitchToBroker(). Because in getCache(), if the broker is used, it will return cache implementation related to broker so that app can clear tokens for related accounts. In addition, according to your comment here, I will change the printStackTrace into logger.